.github/workflows/trigger-release-candidate-build-and-deploy.yml

```diff
@@ -0,0 +1,39 @@
+name: Trigger release candidate build and deploy
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  merge:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout main branch and staging branch
+        uses: actions/checkout@v2
+        with:
+          ref: main
+          fetch-depth: 0 # fetch all history to make merges work correctly
```

Comment:
We set fetch-depth to 1 in order to prevent the checkout step from taking upwards of 10 minutes (our repo is super bloated and fetching all the commits takes forever). Is it possible to figure out how to fetch only the commits that are needed?

Reply:
@atarashansky - I _think_ it is possible. Perhaps there is a way to use `git fetch --shallow-since`, but I am not sure yet.

But since this work is towards an MVP of continuous delivery to `staging`, I don't think this PR needs to be blocked on the fetch-depth issue. If it is taking too long, we can always turn off this workflow.

There are a couple of challenges with figuring out the issue of fetching only the commits from the incoming `main` branch that is:

1. Testing is pretty onerous for GHA stuff. As far as my knowledge is goes with GHA, I would have to keep pushing to this PR to demonstrate a test with test branches. Just look at the number of commits I had to make just to demonstrate that correctness of auto-merge

2. Concurrency poses problems. Say you have multiple commits into `main` in rapid succession that trigger this merge to `staging`. From my reading, there is no order guarantee of how these commits are processed. I don't think something like `fetch-depth=x` will work in this case. The fetching scheme under concurrency needs to be tested for correctness and the more complex the fetching scheme is, the harder it is to test, and the less sure we are about its correctness under concurrency. `fetch-depth=0` does not have this problem under concurrency.

My suggestion is to tackle the fetching scheme in a subsequent PR rather than hold this back for an MVP.

Reply:
OK - sounds good!

Reply:
agreed, reasonble to take on later. can you add an issue to the [post-mvp](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/gh/chanzuckerberg/single-cell-data-portal/5457) epic?

Reply:
some possible solutions usings --merge-base and --deepen options, referenced in these [search results](https://www.google.com/search?q=git+clone+sufficient+history+to+merge+two+branches), but no clear immediate win.

Reply:
Another way entirely around merging would be to destroy and recreate the staging branch every time.

Reply:
@atolopko-czi - I have created a ticket here to capture the fact that we might want to improve the speed of a `stage` deploy: https://github.com/chanzuckerberg/single-cell-data-portal/issues/5896